File: src/main/java/com/rtkit/upsource_manager/payload/api/user/UserInfoResponseDTO.java

package com.rtkit.upsource_manager.payload.api.user;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.rtkit.upsource_manager.payload.api.ABaseUpsourceResponse;

import java.util.List;

@JsonInclude(JsonInclude.Include.NON_NULL)

public class UserInfoResponseDTO extends ABaseUpsourceResponse {

    @JsonProperty("infos")
    private List<Info> infos = null;

    /**
     * No args constructor for use in serialization
     */
    public UserInfoResponseDTO() {
    }

    /**
     * @param infos
     */
    public UserInfoResponseDTO(List<Info> infos) {
        super();
        this.infos = infos;
    }

    @JsonProperty("infos")
    public List<Info> getInfos() {
        return infos;
    }

    @JsonProperty("infos")
    public void setInfos(List<Info> infos) {
        this.infos = infos;
    }

}

File: src/main/java/com/rtkit/upsource_manager/payload/api/UserInfoResponseDTO.kt
package com.rtkit.upsource_manager.payload.api

import com.rtkit.upsource_manager.payload.pacer.review.Participant
import com.rtkit.upsource_manager.payload.pacer.review.ParticipantRoot

class UserInfoResponseDTO(
    response: String
) : ABaseUpsourceResponse(response){

    private val participantRootObj = objectMapper.readValue(response, ParticipantRoot::class.java)

    fun getParticipants(): MutableList<Participant> {
        return participantRootObj.getResult().infos
    }

    fun getFirstParticipant() : Participant {
        return participantRootObj.getResult().infos[0]
    }
}
